react/src/lib/components/DeckGLMap/layers/map/mapLayer.stories.tsx

```diff
@@ -935,3 +940,98 @@ ColorMapRange.parameters = {
         },
     },
 };
+
+// Map layer with color colorselector
+
+const MapLayerColorSelectorTemplate: ComponentStory<typeof DeckGLMap> = (
+    args
+) => {
+    const [colorName, setColorName] = React.useState("Rainbow");
+    const [colorRange, setRange] = React.useState();
+    const [isAuto, setAuto] = React.useState();
+    const [breakPoints, setBreakPoint] = React.useState();
+    const [isLog, setIsLog] = React.useState(false);
+    const [isNearest, setIsNearest] = React.useState(false);
+
+    // user defined breakpoint(domain)
+    const userDefinedBreakPoint = React.useCallback((data) => {
+        if (data) setBreakPoint(data.colorArray);
+    }, []);
+
+    // Get color name from color selector
+    const colorNameFromSelector = React.useCallback((data) => {
+        setColorName(data);
+    }, []);
+
+    // user defined range
+    const userDefinedRange = React.useCallback((data) => {
+        if (data.range) setRange(data.range);
+        setAuto(data.isAuto);
+    }, []);
+
+    // Get interpolation method from color selector to layer
+    const getInterpolateMethod = React.useCallback((data) => {
+        setIsLog(data.isLog);
+        setIsNearest(data.isNearest);
+    }, []);
+
+    // color map function
+    const colorMapFunc = React.useCallback(() => {
+        return createColorMapFunction(colorName, isLog, isNearest, breakPoints);
+    }, [colorName, isLog, isNearest, breakPoints]);
+
+    const min = 100;
+    const max = 1000;
+
+    const updatedLayerData = [
+        {
+            ...meshMapLayerFloat32,
+            colorMapName: colorName,
+            colorMapRange:
+                colorRange && isAuto == false ? colorRange : [min, max],
+            colorMapFunction: colorMapFunc(),
+        },
+    ];
+    return (
+        <DeckGLMap {...args} layers={updatedLayerData}>
+            {
+                // @ts-expect-error This is demonstrated to work with js, but with ts it gives error
+                <View id="view_1_2D">
+                    <div style={{ marginTop: 50 }}>
+                        <ColorLegend
+                            min={min}
+                            max={max}
+                            colorNameFromSelector={colorNameFromSelector}
```

Comment:
Looks very nice!

Suggested improvement: would it be possible to generate the current colorMapFunction from the ColorLegend component?